runtime/src/utils/buffer/write.rs

```diff
@@ -142,6 +142,12 @@ impl<B: Blob> Write<B> {
             })),
         }
     }
+
+    /// Returns the current position of the buffer in the underlying [Blob].
+    pub async fn position(&self) -> u64 {
```

Comment:
It turns out we already handle the truncation case correctly in the write buffer and removing all notion of this handling from `journal` is much cleaner.